### PR TITLE
upgrade ClimaCore, switch to spacefilling curve

### PR DESCRIPTION
**Pull request by simonbyrne**
# PULL REQUEST

## Purpose and Content
upgrade ClimaCore, switch to spacefilling curve

## Benefits and Risks
hopefully better performance

## Linked Issues
(Provide references to any link issues. Use closes #issuenum to automatically close an open issue)
- Fixes #
- Closes #

## PR Checklist
- [ ] This PR has a corresponding issue OR is linked to an SDI.
- [x] I have followed CliMA's codebase [contribution](https://clima.github.io/ClimateMachine.jl/latest/Contributing/) and [style](https://clima.github.io/ClimateMachine.jl/latest/DevDocs/CodeStyle/) guidelines OR N/A.
- [x] I have followed CliMA's [documentation policy](https://github.com/CliMA/policies/wiki/Documentation-Policy).
- [x] I have checked all issues and PRs and I certify that this PR does not duplicate an open PR.
- [x] I linted my code on my local machine prior to submission OR N/A.
- [x] Unit tests are included OR N/A.
- [x] Code used in an integration test OR N/A.
- [x] All tests ran successfully on my local machine OR N/A.
- [x] All classes, modules, and function contain docstrings OR N/A.
- [x] Documentation has been added/updated OR N/A.

### CODE REVIEW

**Comment by simonbyrne** [comment]
This does appear to improve scaling performance. @szy21 let me know how the results look.

**Comment by jiahe23** [comment]
> This does appear to improve scaling performance. @szy21 let me know how the results look.

"Baroclinic wave looks fine but held suarez seems a bit weird. However, the buildkite short run may not generate sensible results for held surarez anyway. I need to run it for 1200 days to see if it looks reasonable, which will take roughly the whole day today. Will keep you two posted." as reply in slack @simonbyrne @sriharshakandala

**Comment by simonbyrne** [comment]
How are you looking at the Held Suarez?

**Comment by szy21** [comment]
@simonbyrne The raw plots in the buildkite look discontinuous (I attached one example), but could it be a problem with the plot function? 
Screen Shot 2022-06-30 at 10 13 42 PM

**Comment by simonbyrne** [comment]
Ah, that's because the plotting code is still assuming the old ordering. Let me think how to fix that.

**Comment by simonbyrne** [comment]
Okay, I updated ClimaCorePlots, so hopefully now it should look correct.

**Comment by szy21** [comment]
It looks ok now, thanks!

**Comment by jiahe23** [comment]
Longrun statistics look reasonal. Here is for dry held suarez.
hs-u
hs-T
hs-theta
hs-T2

**Comment by jiahe23** [comment]
I think this is good to be merged in.

**Comment by simonbyrne** [comment]
Oh I'm away from my computer. Can someone fix the tests?

**Comment by szy21** [comment]
#575 is merged, so this PR only needs to change examples/common_spaces.jl?

File: examples/Manifest.toml [hunk 1]
```diff
@@ -53,6 +53,12 @@ git-tree-sha1 = "a1e2cf6ced6505cbad2490532388683f1e88c3ed"
 uuid = "dd5226c6-a4d4-4bc7-8575-46859f9c95b9"
 version = "0.1.0"
 
+[[deps.ArrayInterfaceStaticArraysCore]]
```
Comment (szy21) [inline examples/Manifest.toml:56]: @simonbyrne Is this change still necessary?

**Comment by szy21** [comment]
@simonbyrne I fixed the mse table, but may have messed up the rebase. Now it doesn't change Manifest.toml. Could you confirm if it's correct? If so feel free to merge it.
